DiaHelperApplication/DiaHelperApplication/Controllers/GlucometerValueViewController.swift

```diff
@@ -0,0 +1,223 @@
+                    if let date = dateFormatter.date(from: text) {
+                        timestamp = date
+                        hideError()
+                    } else {
+                        showError(message: "Time has to be in HH:mm format")
+                        return
+                    }
+                }
+            case 1:
+                if let text = getCellTextFieldText(indexPath: indexPath), let glucose1 = Double(text), glucose1 >= 2 && glucose1 <= 30 {
+                    bloodSugar = glucose1
+                    hideError()
+                } else {
+                    showError(message: "Invalid input for blood sugar.")
+                    return
+                }
+            default:
+                break
+            }
+        }
+        
+        let glucometerBloodSugarTest = GlucometerBloodSugarTest(timestamp: timestamp, bloodSugar: bloodSugar)
+        
+        UserManager.shared.addGlucometerBloodSugarTest(glucometerBloodSugarTest)
+        
+        //print(UserManager.shared.getAllGlucometerBloodSugarTests())
+        delegate?.didSubmitGlucometerTest(glucometerBloodSugarTest.bloodSugar)
+        
+        dismiss(animated: true)
+    }
+    
+    private func getCellTextFieldText(indexPath: IndexPath) -> String? {
+        guard let cell = tableView.cellForRow(at: indexPath) as? NutritionTableViewCell else {
+            return nil
+        }
+        return cell.rightTextField.text
+    }
+    
+    private func setupErrorLabel() {
+        view.addSubview(errorField)
+
+        errorField.translatesAutoresizingMaskIntoConstraints = false
+        NSLayoutConstraint.activate([
+            errorField.leadingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.leadingAnchor, constant: 16),
+            errorField.trailingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.trailingAnchor, constant: -16),
+            errorField.topAnchor.constraint(equalTo: tableView.bottomAnchor, constant: 8),
+        ])
+    }
+
+    private func showError(message: String) {
+        errorField.text = message
+        errorField.isHidden = false
+    }
+
+    private func hideError() {
+        errorField.text = nil
+        errorField.isHidden = true
+    }
+}
+
+extension GlucometerValueViewController: UITableViewDataSource {
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return 2
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: "Cell", for: indexPath) as! NutritionTableViewCell
+        
+        switch indexPath.row {
+        case 0:
+            cell.leftLabel.text = "Time"
+            cell.rightTextField.placeholder = getCurrentTime()
+            cell.rightTextField.delegate = self
+        case 1:
+            cell.leftLabel.text = "Enter blood sugar 🩸"
+            cell.rightTextField.placeholder = "Enter Value"
```

Comment:
This text field should be set as numerical so it would only show a numbers keypad with `textField.keyboardType = .numberPad`